io/io/inc/dcqueue.h

```diff
@@ -0,0 +1,212 @@
+//===----------------------------------------------------------------------===//
```

Comment:
Well, this is the code as imported. I wanted to give correct authorship to it, and then make my modifications on top. I will put everything in ROOT style as part of the work.